File: targets/js-ts-express/services/toolService.ts
// Diagnostic command logic; the target is staged in a field between calls.
const { exec } = require('child_process');

class ToolService {
  private target: string = '';

  stageTarget(host: string): void {
    this.target = host;
  }

  // Runs ping against the staged target.
  runStagedDiag(cb: (err: unknown, out: string) => void): void {
    exec('ping -c 1 ' + this.target, (err, stdout) => cb(err, stdout));
  }
}

export const toolService = new ToolService();

File: targets/csharp-aspnet/Services/ToolService.cs
using System.Diagnostics;

namespace BadDemo.Services
{
    /// <summary>Diagnostic command logic; the target is stored in a field between calls.</summary>
    public class ToolService
    {
        /// <summary>Target staged by the controller between calls.</summary>
        private string _target = "";

        public void StageTarget(string host)
        {
            _target = host;
        }

        /// <summary>Runs the ping diagnostic against the staged target.</summary>
        public int RunStagedDiag()
        {
            var proc = Process.Start("cmd.exe", "/c ping " + _target);
            proc.WaitForExit();
            return proc.ExitCode;
        }
    }
}
